pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -281,10 +281,14 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.CloudupMode
 
 	if ig.Spec.InstanceMetadata != nil && ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit != nil {
 		lt.HTTPPutResponseHopLimit = ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit
+	} else if ig.IsControlPlane() && (b.Cluster.IsKubernetesLT("1.27") || !(b.Cluster.Spec.IAM != nil && fi.ValueOf(b.Cluster.Spec.IAM.UseServiceAccountExternalPermissions))) {
```

Comment:
Can we move this to a function? My head hurts a bit from trying to read this expression :)

Reply:
Is there a good reason to not use IRSA for k8s < 1.27 (see https://github.com/kubernetes/kops/pull/14572)?

Reply:
I think latest change makes it a little more readable. Even if it were a function it would still be hard to even name it. :))